Impl viewsource

glacambre opened the pull request:
I  use Pentadactyl's `:viewsource` command from time to time. This is a re-implementation of this command. I added a `viewsource` setting which can be set either to `tridactyl`or to `default` in order to alleviate the pain of Tridactyl not loading on Firefox's `view-source` pages.
Fun fact, when `viewsource` is set to `default`, you can `:viewsource file:///home/user/file`. It works quite well with plaintext files and pictures, although Tridactyl obviously doesn't load on these pages.

bovine3dom commented:
Hi, thanks for doing this. There are a couple of issues:

On news.bbc.co.uk, `viewsource` leaves the BBC News header on the page.

I'm not sure I approve of the grey background, but it does not cover the whole page, which can be seen if you scroll to the right.

glacambre commented:
Rebased, fixed viewsource leaving the header and made sure the viewsource element respects the user's theme and fills the whole page's width.

I hope there's a limit to the number of bugs I can introduce in such a small amount of code and that these were the last :)

cmcaine commented:
Some issues:

viewsource element doesn't always cover the whole of the window: http://students.cs.manchester.ac.uk/~mbax9cc4/

viewsource element appears at end of page on newtab page.

viewsource element doesn't disappear if you go back in history on a SPA: http://compsoc.man.ac.uk/~cmc/biol10232/
